build.ps1

```diff
@@ -0,0 +1,466 @@
+# Default to Debug
+$Configuration = 'Debug'
+
+# Color
+$Success = 'Green'
+$Warning = 'Yellow'
+$Err = 'Red'
+
+Function Error ($msg)
+{
+    Write-Host "[Error:]" $msg -ForegroundColor $Err
+}
+
+Function Warning ($msg)
+{
+    Write-Host "[Warning:]" $msg -ForegroundColor $Warning
+}
+
+Function Success ($msg)
+{
+    Write-Host "[Success:]" $msg -ForegroundColor $Success
+}
+
+Function Info ($msg)
+{
+    Write-Host "[Info:]" $msg
+}
+
+if ($args.Count -eq 0)
+{
+    $TestType = 'Nightly'
+    $Configuration = 'Release'
+}
+elseif ($args[0] -match 'quick' -or ($args[0] -match '-q')) 
+{
+    $TestType = "Quick"
+}
+elseif ($args[0] -match 'DisableSkipStrongName')
+{
+    $TestType = "DisableSkipStrongName"
+}
+elseif ($args[0] -match 'EnableSkipStrongName')
+{
+    $TestType = "EnableSkipStrongName"
+}
+else 
+{
+    Error("Unknown input ""$args"". It can be empty or ""quick|DisableSkipStrongName|EnableSkipStrongName"".")
+    exit
+}
+
+$Build = 'build'
+if ($args -contains 'rebuild')
+{
+    $Build = 'rebuild'
+}
+
+$PROGRAMFILESX86 = [Environment]::GetFolderPath("ProgramFilesX86")
+$env:ENLISTMENT_ROOT = Split-Path -Parent $MyInvocation.MyCommand.Definition
+$ENLISTMENT_ROOT = Split-Path -Parent $MyInvocation.MyCommand.Definition
+$LOGDIR = $ENLISTMENT_ROOT + "\bin"
+
+# Default to use Visual Studio 2015
+$VS14MSBUILD=$PROGRAMFILESX86 + "\MSBuild\14.0\Bin\MSBuild.exe"
+$VSTEST = $PROGRAMFILESX86 + "\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
+$FXCOPDIR = $PROGRAMFILESX86 + "\Microsoft Visual Studio 14.0\Team Tools\Static Analysis Tools\FxCop"
+$SN = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\sn.exe"
+$SNx64 = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\x64\sn.exe"
+
+# Use Visual Studio 2017 compiler for .NET Core and .NET Standard. Because VS2017 has different paths for different
+# versions, we have to check for each version. Meanwhile, the dotnet CLI is required to run the .NET Core unit tests in this script.
+$VS15VERSIONS = "Enterprise", "Professional", "Community"
+$VS15MSBUILD = $null
+ForEach ($version in $VS15VERSIONS)
+{
+    $tempMSBuildPath = ($PROGRAMFILESX86 + "\Microsoft Visual Studio\2017\{0}\MSBuild\15.0\Bin\MSBuild.exe") -f $version
+    if([System.IO.File]::Exists($tempMSBuildPath))
+    {
+        $VS15MSBUILD = $tempMSBuildPath
+        break
+    }
+}
+
+$DOTNETDIR = "C:\Program Files\dotnet\"
+$DOTNETTEST = $null
+if ([System.IO.File]::Exists($DOTNETDIR + "dotnet.exe"))
+{
+    $DOTNETTEST = $DOTNETDIR + "dotnet.exe"
+}
+else
+{
+   Error("The dotnet CLI must be installed to run any .NET Core tests.")
+   exit
+}
+
+# Other variables
+$FXCOP = $FXCOPDIR + "\FxCopCmd.exe"
```

Comment:
Should FXCOPDIR for VS2017 be Microsoft Visual Studio\2017\Enterprise\Team Tools\Static Analysis Tools\FxCop? thought?

Reply:
That's a good point. I think the right thing to do is align Fxcop locally with Fxcop in the build. Given that the output is ignored in this script as it is, I'm going to remove it to enable this checkin. A future item can be added to resolve the local and CI versions of Fxcop. Sound good?

Reply:
Sound reasonable to me, since this shouldn't be a blocker.